pkg/util/blockutil/block_time_calculator.go

```diff
@@ -0,0 +1,43 @@
+package blockutil
+
+import "time"
+
+type (
+	// BlockTimeCalculator calculates block time of a given height.
+	BlockTimeCalculator struct {
+		getBlockInterval    getBlockIntervalFn
+		getTipHeight        getTipHeightFn
+		getHistoryBlockTime getHistoryblockTimeFn
+	}
+
+	getBlockIntervalFn    func(uint64) time.Duration
+	getTipHeightFn        func() uint64
```

Comment:
get block time func may return error, should define get tip height `(uint64, error)` as well?

Reply:
Currently, the method `TipHeight() uint64` of the blockchain does not return an error. We can keep it this way.